docs/en/operation/Journal.md

```diff
@@ -93,3 +94,49 @@ If the restore succeeds, you should see a log message along the lines of
 INFO AlluxioMasterProcess - Restored 57 entries from backup
 ```
 in the primary master logs.
+
+### Changing Masters
+
+If the journal is stored in a distributed system like HDFS, changing masters is easy.
+As long as the new master sets `alluxio.master.journal.folder` the same as the old
+master, it will start up in the same state that the old master left off.
+
+If the journal is stored on the local filesystem, the journal folder needs to be
+copied to the new master.
+
+## Advanced
+
+### Journal Checkpoints
+
+When running with a single master, the journal folder size will grow indefinitely
+as metadata operations are written to journal log files. To address this, production
+deployments should run in HA mode with multiple Alluxio masters. The non-primary
+masters will create checkpoints of the master state and clean up the logs that
+were written prior to the checkpoints. For example, if 3 million files were
+created and then 2 million were deleted, the journal logs would contain 5 million
+total entries. Then if a checkpoint is taken, the checkpoint will contain only
+1 million entries, and the original 5 million entries will be deleted.
+
+By default, checkpoints are taken every 2 million entries. This can be configured by
+setting `alluxio.master.journal.checkpoint.period.entries` on the masters. Setting
+the value lower will reduce the amount of disk space needed by the journal at the
+cost of additional work for the secondary masters.
+
+If HA mode is not an option, it is possible to run two masters on the same node,
+where the second master exists only to write checkpoints. In this setup, both
```

Comment:
How does this work? If the primary dies, this secondary will NOT become master?

Reply:
Right, SecondaryMaster doesn't participate in leader election. I'll add a mention that it won't take over in case of primary failure